Source/Fuse.Reactive.JavaScript/TreeObservable/TreeObject.uno

```diff
@@ -0,0 +1,111 @@
+using Uno.Collections;
+using Uno;
+
+namespace Fuse.Reactive
+{
+	class TreeObject : ObjectMirror, IObservableObject
+	{
+		/** Does not poulate the _props. Must call Set() later */
+		protected TreeObject(Scripting.Object obj) : base(obj) {}
+
+		public IPropertySubscription Subscribe(IPropertyObserver observer)
+		{
+			return new PropertySubscription(this, observer);
+		}
+
+		internal class PropertySubscription : Subscription, IPropertySubscription
+		{
+			readonly IPropertyObserver _observer;
+			
+			public PropertySubscription(TreeObject om, IPropertyObserver observer): base(om)
+			{
+				_observer = observer;
+			}
+
+			class JSThreadSet
+			{
+				Scripting.Object _obj;
+				string _key;
+				object _value;
+				public JSThreadSet(Scripting.Object obj, string key, object value)
+				{
+					_obj = obj;
+					_key = key;
+					_value = value;
+				}
+				public void Perform()
+				{
+					if (_obj.ContainsKey("__fuse_requestChange")) {
+						((Scripting.Function)_obj["__fuse_requestChange"]).Call(_key, _value);
+					}
+					else
+					{
+						_obj[_key] = _value;
+					}
+				}
+			}
+
+			bool IPropertySubscription.TrySetExclusive(string key, object newValue)
+			{
+				var t = SubscriptionSubject as TreeObject;
```

Comment:
Please do
```uno
var t = (TreeObject)SubscriptionSubject;
```

This is clearer. We're going to be dereferencing `t`  on the next statement anyway, so it needs to be non-null. Doing a proper cast gives us a better error-message if this was every wrong.